### PR TITLE
[DOCS] Archived settings block other setting updates

### LINKED ISSUE #61175: PUT /_cluster/settings/" while throw IllegalArgumentException
**Elasticsearch version** : 6.8.11
**Plugins installed**: []
**JVM version** : 1.8.0_45
**OS version** : GNU/Linux

**Description of the problem including expected versus actual behavior**:
First the cluster version is 5.6.8,  then we upgrade the cluster to 6.8.11. If we update cluster settings in the new cluster, it will throw the IllegalArgumentException:
```
"reason": "unknown setting [archived.indices.ttl.interval] please check that any required plugins are installed, or check the breaking changes documentation for removed settings"

```
The reason is clear, there exists a setting in the ES5 cluster, which is archived in ES6 cluster. After upgrading to ES6, if we update the cluster setting , ES6 server will validate the archived settings:`indices.ttl.interval` and not ignore it, which will throw exception.
https://github.com/elastic/elasticsearch/blob/0f256948a5176ac4baab0a217e1107c0f86063a5/server/src/main/java/org/elasticsearch/action/admin/cluster/settings/SettingsUpdater.java#L94
As the result:
1.There doesn’t exist api which can be used to delete those cluster settings.
2.I will be unable to update any cluster settings in ES6.

**Steps to reproduce**:
step 1:
Push the cluster setting to ES5.6.8 cluster:
```
PUT _cluster/settings/admin/lc_gh_4
{
    "persistent": {
        "indices.ttl.interval":"6000s"
    }
}
```
step 2:
Upgrade the cluster to 6.8.11
step 3:
Push the cluster setting to the cluster:
```
PUT _cluster/settings
{
     "persistent": {
      "cluster": {
         "routing": {
            "allocation": {
               "cluster_concurrent_rebalance": "3"
            }
         }
      }
     }
}
```
ES will throw the exception
```
 "reason": "unknown setting [archived.indices.ttl.interval] please check that any required plugins are installed, or check the breaking changes documentation for removed settings"
```

### PR DESCRIPTION
**Pull request by jrodewig**
**Changes:**

* Notes that archived cluster settings block cluster setting updates. Previously, the docs stated that ES ignored archived cluster settings.
* Notes that archived index settings can block index settings updates. For example, it blocks `index.hidden` but not `number_of_replicas`. Previously, the docs stated that you could safely ignore archived index settings.

### Preview
https://elasticsearch_82069.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/archived-settings.html

Relates #78351

Closes #61175

### CODE REVIEW

File: docs/reference/upgrade/archived-settings.asciidoc [hunk 1]
```diff
@@ -48,18 +52,23 @@ templates during an upgrade. Attempts to use a template that contains an
...
 ----
-GET my-index/_settings?flat_settings=true&filter_path=**.settings.archived*
+GET */_settings?flat_settings=true&filter_path=**.settings.archived*
 ----
 // TEST[s/^/PUT my-index\n/]
 
-Removing archived index settings requires a reindex after the upgrade. However,
-reindexing can be resource intensive. Because {es} ignores archived settings,
-you can safely leave them in place if wanted.
+Removing an index's archived index settings requires a <<docs-reindex,reindex>>.
```
Comment (j-bennet) [inline docs/reference/upgrade/archived-settings.asciidoc:69]: This clarification is great.

**Review by j-bennet — APPROVED** [review]
Looks great! 👍  I really hope the inconsistent behavior can be fixed in the future though, besides only being clarified in the documentation.

**Comment by jrodewig** [comment]
@DaveCTurner @gwbrown Do one of you mind taking a look at this PR at your convenience? Thank you!

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Review by AthenaEryma — APPROVED** [review]
LGTM, thanks! And again, apologies for the delay!

**Comment by jrodewig** [comment]
Thanks @gwbrown!

**Comment by jrodewig** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.0|[](https://github.com/elastic/elasticsearch/pull/83172)|
|✅|7.17|[](https://github.com/elastic/elasticsearch/pull/83173)|
|✅|7.16|[](https://github.com/elastic/elasticsearch/pull/83174)|


### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**